### PR TITLE
added check in tags for services in catalog

### PR DESCRIPTION
**Pull request by girishramnani**
## What is the purpose of this change? What does it change?
This change checks for `hidden` tag in the services and hides them.

## Was the change discussed in an issue?
partly https://github.com/redhat-developer/odo/issues/1180

## How to test changes?
`odo catalog list services`
`odo catalog search service <service>`

### CODE REVIEW

File: pkg/occlient/occlient.go [hunk 1]
```diff
@@ -1936,16 +1938,18 @@ func (c *Client) GetClusterServiceClass(serviceName string) (*scv1beta1.ClusterS
 		FieldSelector: fields.OneTermEqualSelector("spec.externalName", serviceName).String(),
 	}
 	searchResults, err := c.serviceCatalogClient.ClusterServiceClasses().List(opts)
+
+	items := filterHiddenClasses(searchResults.Items)
```
Comment (geoand) [inline pkg/occlient/occlient.go]: To be honest, I don't think this should be done. 
From what I understand, `hidden` is meant only to hide things when all services are listed, not when obtaining info in other ways

File: pkg/catalog/catalog.go [hunk 2]
```diff
@@ -138,22 +137,18 @@ func getDefaultBuilderImages(client *occlient.Client) ([]CatalogImage, error) {
...
-
 	var builderImages []CatalogImage
 
 	// Get builder images from the available imagestreams
 	for _, imageStream := range imageStreams {
 		var allTags []string
 		buildImage := false
-
 		for _, tag := range imageStream.Spec.Tags {
-
```
Comment (cdrage) [inline pkg/catalog/catalog.go]: @girishramnani you you need to revert these changes, you've deleted the newlines

File: pkg/occlient/utils.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package occlient
+
+import scv1beta1 "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+
+func isHidden(tags []string) bool {
```
Comment (metacosm) [inline pkg/occlient/utils.go]: I'd actually make this function a little more generic: `hasTag(tags []string, tag string)` instead of making it specific to `hidden`. Asking selfishly because I actually need such a function ;)

Comment (girishramnani) [reply pkg/occlient/utils.go]: ack

**Comment by geoand** [comment]
We could have an integration test, using one of the `sso` services that should be hidden (don't remember which one it is, @jorgemoralespou should now)

File: tests/e2e/service_test.go [hunk 4]
```diff
@@ -81,6 +83,14 @@ var _ = Describe("odoServiceE2e", func() {
...
 			})
 		})
+
+		Context("odo hides a hidden service in service catalog", func() {
+			It("not show a hidden service in the catalog", func() {
+				runCmd("oc apply -f https://github.com/openshift/library/raw/master/official/sso/templates/sso72-https.json -n openshift")
+				output := runFailCmd("odo catalog search service sso72-https")
+				Expect(output).To(ContainSubstring("No service matched the query: sso72-https"))
+			})
+		})
```
Comment (amitkrout) [inline tests/e2e/service_test.go]: +1

**Comment by girishramnani** [comment]
Merging
